notebooks/30_conditionals/exkurs_methoden.ipynb

```diff
@@ -0,0 +1,104 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Exkurs: Methoden\n",
+    "Dieser Exkurs (Exkurs = Sie müssen das nicht unbedingt durcharbeiten. Allerdings helfen die Erklärungen möglicherweise, eine besseres Verständnis zu bekommen.) beschäftigt sich mit dem Begriff Methoden.\n",
+    "\n",
+    "In den bisherigen Kapiteln tauchten die Begriffe Funktion und Methoden auf. Beispielsweise gibt es die Funktion `print()` oder die String-Methode `.isdigit()`. Funktionen werden noch detailliert besprochen werden. Im Augenblick nur kurz: Funktionen \"machen\" irgendetwas. Z.B. druckt die Funktion `print()` die übergebenen Argument aus. Methoden sind so ähnlich wie Funktionen, allerdings hängen die Methoden von Objekten ab. `isdigit()` hängt z.B. an dem Objekt String.\n",
+    "## Objekt-Orientierung\n",
+    "In den 80'er und 90'er Jahren wurde Software immer größer und komplexer. War zuvor die Hardware das Bottleneck bei der Systementwicklung, wurde jetzt die Komplexität der Software immer mehr zum limitierenden Faktor. Um Software besser programmierbar und wartbar zu machen, wurde die *objekt-orientierte* Softwareentwicklung eingeführt.\n",
```

Comment:
Ich glaube das stimmt so nicht. Smalltalk wurde in den 1970er entwickelt. Smalltalk gilt als die erste OO Sprache. Somit denke ich, dass die Ursprünge der OO früher liegen.